File: server/src/services/storeService.ts
import { Store, CreateStoreRequest, UpdateStoreRequest } from '../types/store';
import { StoreModel } from '../models/StoreModel';
import { createError } from '../middleware/errorHandler';

export class StoreService {

  async getAllStores(): Promise<Store[]> {
    try {
    return await StoreModel.findAll();
    } catch (error) {
      console.error('StoreService.getAllStores error:', error);
      // Return empty array on any error to prevent 500
      return [];
    }
  }

  async getStoresByCategory(category: string): Promise<Store[]> {
    try {
    if (category === 'All') {
      return await StoreModel.findAll();
    }
    return await StoreModel.findByCategory(category);
    } catch (error) {
      console.error('StoreService.getStoresByCategory error:', error);
      return [];
    }
  }

  async searchStores(query: string): Promise<Store[]> {
    try {
    return await StoreModel.search(query);
    } catch (error) {
      console.error('StoreService.searchStores error:', error);
      return [];
    }
  }

  async getStoreById(id: string): Promise<Store> {
    const store = await StoreModel.findById(id);
    if (!store) {
      throw createError(`Store with ID ${id} not found`, 404);
    }
    return store;
  }

  async createStore(storeData: CreateStoreRequest): Promise<Store> {
    return await StoreModel.create(storeData);
  }

  async updateStore(id: string, updateData: UpdateStoreRequest): Promise<Store> {
    const store = await StoreModel.update(id, updateData);
    if (!store) {
      throw createError(`Store with ID ${id} not found`, 404);
    }
    return store;
  }

  async deleteStore(id: string): Promise<void> {
    const deleted = await StoreModel.delete(id);
    if (!deleted) {
      throw createError(`Store with ID ${id} not found`, 404);
    }
  }

  async getCategories(): Promise<string[]> {
    try {
    return await StoreModel.getCategories();
    } catch (error) {
      console.error('StoreService.getCategories error:', error);
      // Return empty array on any error to prevent 500
      return [];
    }
  }

}

File: server-java/src/main/java/com/invictusmall/service/StoreService.java
package com.invictusmall.service;

import com.invictusmall.model.Store;
import com.invictusmall.repository.StoreRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class StoreService {

    @Autowired
    private StoreRepository storeRepository;

    public List<Store> getAllStores(String category, String search) {
        if (category != null && !category.isEmpty()) {
            return storeRepository.findByCategory(category);
        }
        if (search != null && !search.isEmpty()) {
            return storeRepository.searchStores(search);
        }
        return storeRepository.findAll();
    }

    public Store getStoreById(String id) {
        return storeRepository.findById(id).orElse(null);
    }

    public List<String> getCategories() {
        return storeRepository.findAllCategories();
    }
}


